README.md

```diff
@@ -586,6 +587,80 @@ Here's an example of how to use the command:
 python src/main.py export-public-keys --data-dir <path-to-data-dir>
 ```
 
+## V4 Upgrade Guide
+
+### Pectra Upgrade Support
+
+Ensure your vault is updated to version 5 for Ethereum network or version 3 for Gnosis network for full Pectra compatibility.
+
+After the upgrade, validator balances are no longer limited to 32 ETH. By default, we register 0x02 validators. To register 0x01 validators, add the flag `--validators-type=0x01`.
+
+When replenishing validators, funds first top up existing 0x02 validators up to 2048 ETH. New validators are registered once the vault accumulates another 32 ETH.
+
+To migrate 0x01 validators to 0x02, use the consolidate command (see [reference](#validators-consolidation)).
+
+Also, partial withdrawals for compound validators are now supported. Partial withdrawals are significantly faster and more efficient than full validator exits. Even full validator exits now can be processed via execution request call.
```

Comment:
Please add description how partial withdrawals work:
- how the validator is selected?
- what happens if all the validators are with 32 ETH?
etc.